### PR TITLE
net: tcp: Retransmit FIN message with right sequence number in FIN1 state

### PR DESCRIPTION
**Pull request by ssharks**
When an ACK is acknowledged in the FIN1 state for the FIN flag, properly increment the ACK counter.
When retransmitting the FIN packet from the FIN1 state, reduce the sequence counter by 1 to avoid sending a "second" FIN flag.

### CODE REVIEW

**Comment by ssharks** [comment]
@AndreyDodonov-EH: Could you review this PR?

File: tests/net/tcp/src/main.c [hunk 1]
```diff
@@ -96,6 +96,7 @@ static struct sockaddr_in6 peer_addr_v6_s = {
 static struct net_if *iface;
```
Comment (jukkar) [inline tests/net/tcp/src/main.c:96]: Please rename the test commit to
`tests: net: tcp: ....`
so that it is easier to search from commit log.

**Review by jukkar — COMMENTED** [review]
LGTM, minor nit about the commit subject.

**Comment by AndreyDodonov-EH** [comment]
> @AndreyDodonov-EH: Could you review this PR?

@ssharks 
As far as I understand, this results in keeping ACK-ing the last actually processed sequence number, instead of ACK-ing new sequence numbers?

**Comment by ssharks** [comment]
> > @AndreyDodonov-EH: Could you review this PR?
> 
> @ssharks As far as I understand, this results in keeping ACK-ing the last actually processed sequence number, instead of ACK-ing new sequence numbers?
> 
> I couldn't find in https://www.rfc-editor.org/rfc/rfc9293 what is the expected behavior in case of half-closed connection. I interepreted it that yes, we ACK only old sequence number, but also RST

This PR only fixes the wrong sequence number in retransmission. I'm working on a testcase for the last trace you send in. The reset behavior is a little harder, so I think we should seperate them in 2 PRs at least.

File: tests/net/tcp/src/main.c (hunk 1 above)
Comment (ssharks) [reply tests/net/tcp/src/main.c:96]: Done

**Comment by ssharks** [comment]
Superseeded by https://github.com/zephyrproject-rtos/zephyr/pull/61353

**Pull request closed without merging**